src/main/java/com/bettercloud/vault/api/Logical.java

```diff
@@ -627,17 +627,23 @@ public LogicalResponse upgrade(final String kvPath) throws VaultException {
     /**
      * <p> Provides which secrets engine version to use for the desired path.
      * Checks to see if the relative path is in the secretsEnginePathMap, and returns the specified value if it is.
+     * Checks secretsEnginePathMap in order, so the first matching entry defined in the map takes precedence.
      * Otherwise return the globalEngineVersion. </p>
      *
      * @param secretPath The desired Vault path to access
      * @return The engine version that should be used to access the path
      */
     private Integer engineVersionForSecretPath(final String secretPath) {
         if (!this.config.getSecretsEnginePathMap().isEmpty()) {
-            Optional<String> matchedKeyPrefix = this.config.getSecretsEnginePathMap().keySet().stream()
-                    .filter(key -> secretPath.startsWith(key))
-                    .findFirst();
-            return matchedKeyPrefix.isPresent() ?
+            Optional<String> matchedKeyPrefix = null;
```

Comment:
instead of `null`, let's change this to `Optional.empty` and adjust line 646 to check for `matchedKeyPrefix.present()`

Reply:
Okay! Sounds good

Reply:
Updated!